torch/distributed/_tensor/sharding_prop.py

```diff
@@ -269,18 +269,19 @@ def __init__(self, propagator: ShardingPropagator) -> None:
 
         # cache table for sharding propagation results, we might need to
         # limit the size of the cache table in the future
-        self.cached_prop_results: Dict[OpSchema, OutputSharding] = {}
+        self.cached_prop_results: Dict[str, OutputSharding] = {}
```

Comment:
it looks like we are changing `op_schema` equal and in the meanwhile we are using str to index  and cache the proper results, I feel this two thing might be mutual exclusive. I feel we should still use hash instead of str

Reply:
No, the op_schema gets changed in propagation. Before we figure out a way to not change it, can we temporarily use str conversion?